lib/bitarray-array.rb

```diff
@@ -43,8 +43,9 @@ def each_byte
   end
 
   # Returns the total number of bits that are set
-  # (The technique used here is about 6 times faster than using each or inject direct on the bitfield)
+  # Use Brian Kernighan's way, see
+  # https://graphics.stanford.edu/~seander/bithacks.html#CountBitsSetKernighan
   def total_set
-    @field.inject(0) { |a, byte| a += byte & 1 and byte >>= 1 until byte == 0; a }
+    @field.each_byte.inject(0) { |a, byte| (a += 1; byte &= byte - 1) while byte > 0 ; a }
```

Comment:
The new method iterates once per each set bit. Consider `byte` with a value of 128. The old method would iterate 8 times but the new method iterates only once.

It's more common to see a less significant speed improvement, of course. If all the bits are set (`255`), it's the same number of iterations. On my tests with about half the bits set for a bloom filter, this code change was about twice the speed.

Reply:
Note also that the old code in version `1.2.0` (but already removed from `master`) called `@field.bytes`; this resulted in an array of bytes and caused substantial memory usage rather than just iterating over the `@field`.